### PR TITLE
net-p2p/amule: EAPI=7 bump + >=net-libs/libupnp-1.8 support

### PR DESCRIPTION
**Pull request by puleglot**

### CODE REVIEW

File: acct-user/amule/amule-0.ebuild [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit acct-user
+
+DESCRIPTION="User for net-p2p/amule"
+ACCT_USER_ID=468
+ACCT_USER_GROUPS=( amule )
+ACCT_USER_HOME="/var/lib/amule"
```
Comment (juippis) [inline acct-user/amule/amule-0.ebuild]: Please verify whether this is needed or not, per new instructions in https://devmanual.gentoo.org/ebuild-writing/users-and-groups/index.html#choosing-a-home-directory

File: net-p2p/amule/amule-2.3.2-r4.ebuild [hunk 2]
```diff
@@ -1,19 +1,19 @@
 # Copyright 1999-2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
+EAPI=7
```
Comment (juippis) [inline net-p2p/amule/amule-2.3.2-r4.ebuild]: Also please don't use `git mv` or edit the previous file, but make a new copy, edit that, commit that+Manifest and if you wish, make a new commit removing the old file+Manifest hash.

File: acct-user/amule/amule-0.ebuild (hunk 1 above)
Comment (puleglot) [reply acct-user/amule/amule-0.ebuild]: I moved home directory management into net-p2p/amule ebuild. Init sctipt sets ${HOME} environment variable that should be enough for the daemon to work properly.

File: net-p2p/amule/amule-2.3.2-r4.ebuild (hunk 2 above)
Comment (puleglot) [reply net-p2p/amule/amule-2.3.2-r4.ebuild]: done

File: net-p2p/amule/amule-2.3.2-r4.ebuild [hunk 3]
```diff
@@ -99,11 +96,36 @@ src_install() {
...
 		newinitd "${FILESDIR}"/amuled.initd amuled
 	fi
 	if use remote; then
-		newconfd "${FILESDIR}"/amuleweb.confd amuleweb
+		newconfd "${FILESDIR}"/amuleweb.confd-r1 amuleweb
 		newinitd "${FILESDIR}"/amuleweb.initd amuleweb
 	fi
+
+	keepdir /var/lib/${PN}
+	fowners amule:amule /var/lib/${PN}
```
Comment (juippis) [inline net-p2p/amule/amule-2.3.2-r4.ebuild]: You depend on acct-user/amule conditionally, yet call this unconditionally. It results in broken ebuild, 
```
chown: invalid user: ‘amule:amule’
 * ERROR: net-p2p/amule-2.3.2-r4::testworld failed (install phase):
 *   fowners failed
 * 
 * If you need support, post the output of `emerge --info '=net-p2p/amule-2.3.2-r4::testworld'`,
 * the complete build log and the output of `emerge -pqv '=net-p2p/amule-2.3.2-r4::testworld'`.
 * The complete build log is located at '/tmp/portage/net-p2p/amule-2.3.2-r4/temp/build.log'.
 * The ebuild environment file is located at '/tmp/portage/net-p2p/amule-2.3.2-r4/temp/environment'.
 * Working directory: '/tmp/portage/net-p2p/amule-2.3.2-r4/work/aMule-2.3.2'
 * S: '/tmp/portage/net-p2p/amule-2.3.2-r4/work/aMule-2.3.2'

>>> Failed to emerge net-p2p/amule-2.3.2-r4
```

File: net-p2p/amule/amule-2.3.2-r4.ebuild [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+EAPI=7
+WX_GTK_VER="3.0-gtk3"
+
+inherit wxwidgets user
+
+MY_P="${PN/m/M}-${PV}"
+
+DESCRIPTION="aMule, the all-platform eMule p2p client"
+HOMEPAGE="http://www.amule.org/"
+SRC_URI="mirror://sourceforge/${PN}/${MY_P}.tar.xz"
```
Comment (juippis) [inline net-p2p/amule/amule-2.3.2-r4.ebuild]: Hmm, I dont think this `mirror://sourceforge:` lists mirrors anymore but points to https://download.sourceforge.net, so you could replace it and we can get rid of the mirror file at some point.

File: net-p2p/amule/amule-2.3.2-r4.ebuild [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+COMMON_DEPEND="
+	dev-libs/boost:=
+	dev-libs/crypto++:=
+	sys-libs/binutils-libs:0=
+	sys-libs/zlib
+	>=x11-libs/wxGTK-3.0.4:${WX_GTK_VER}[X?]
+	stats? ( media-libs/gd:=[jpeg,png] )
+	geoip? ( dev-libs/geoip )
+	upnp? ( net-libs/libupnp:* )
+	remote? ( media-libs/libpng:0= )
```
Comment (juippis) [inline net-p2p/amule/amule-2.3.2-r4.ebuild]: Please sort the unconditional list alphabetically.

File: net-p2p/amule/amule-2.3.2-r4.ebuild [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+"
+DEPEND="${COMMON_DEPEND}"
+RDEPEND="${COMMON_DEPEND}"
```
Comment (juippis) [inline net-p2p/amule/amule-2.3.2-r4.ebuild]: Doesn't look like `COMMON_DEPEND` is needed, as DEPEND and RDEPEND are no different.

File: net-p2p/amule/amule-2.3.2-r4.ebuild [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+	"${FILESDIR}/${PN}-2.3.2-disable-version-check.patch"
+	"${FILESDIR}/${PN}-2.3.2-fix-crash-when-shared-files-changed.patch"
+)
+
+pkg_setup() {
+	if use stats && ! use X; then
+		einfo "Note: You would need both the X and stats USE flags"
+		einfo "to compile aMule Statistics GUI."
+		einfo "I will now compile console versions only."
+	fi
```
Comment (juippis) [inline net-p2p/amule/amule-2.3.2-r4.ebuild]: Implement `REQUIRED_USE` for this?

File: net-p2p/amule/amule-2.3.2-r4.ebuild (hunk 5 above)
Comment (puleglot) [reply net-p2p/amule/amule-2.3.2-r4.ebuild]: This gets sorted in a later commit (switch to acct-{user,group}).

File: net-p2p/amule/amule-2.3.2-r4.ebuild (hunk 7 above)
Comment (puleglot) [reply net-p2p/amule/amule-2.3.2-r4.ebuild]: I don't think `REQUIRED_USE` it is needed here. USE="stats -X" is a valid use-flag combination. Maybe this warning should be removed.

**Pull request closed without merging**